src/Compilers/CSharp/Test/Semantic/Semantics/StaticNullChecking.cs

```diff
@@ -5682,178 +5693,289 @@ void M(string? s)
+                // (8,9): warning CS8602: Possible dereference of a null reference.
+                //         s.ToString(); // warn 2
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "s").WithLocation(8, 9)
+                );
         }
-        else
+
+        [Fact]
+        public void MethodWithRefNullableParameter_WithNonNullableRefArgument()
         {
-            s.ToString(); // warn 2
+            CSharpCompilation c = CreateCompilation(@"
+public class C
+{
+    public void Main(string s)
+    {
+        M(ref s); // warn 1
+        s.ToString(); // warn 2
+    }
+    public static void M(ref string? value) => throw null;
+}
+", parseOptions: TestOptions.Regular8);
+
+            c.VerifyDiagnostics(
+                // (6,15): warning CS8600: Converting null literal or possible null value to non-nullable type.
+                //         M(ref s); // warn 1
+                Diagnostic(ErrorCode.WRN_ConvertingNullableToNonNullable, "s").WithLocation(6, 15),
+                // (7,9): warning CS8602: Possible dereference of a null reference.
+                //         s.ToString(); // warn 2
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "s").WithLocation(7, 9)
+                );
         }
+        [Fact]
+        public void MethodWithRefNullableParameter_WithNonNullableLocal()
+        {
+            CSharpCompilation c = CreateCompilation(@"
+public class C
+{
+    public void Main()
+    {
+        string s = ""hello"";
+        M(ref s); // warn 1
+        s.ToString(); // warn 2
     }
-    static bool Method([NotNullWhenFalse] string? s, string? s2) => throw null;
+    public static void M(ref string? value) => throw null;
 }
-" + NotNullWhenFalseAttributeDefinition, parseOptions: TestOptions.Regular8);
+", parseOptions: TestOptions.Regular8);
 
             c.VerifyDiagnostics(
-                // (9,13): warning CS8602: Possible dereference of a null reference.
-                //             s.ToString(); // warn 1
-                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "s").WithLocation(9, 13),
-                // (13,13): warning CS8602: Possible dereference of a null reference.
-                //             s.ToString(); // warn 2
-                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "s").WithLocation(13, 13)
+                // (7,15): warning CS8600: Converting null literal or possible null value to non-nullable type.
+                //         M(ref s); // warn 1
+                Diagnostic(ErrorCode.WRN_ConvertingNullableToNonNullable, "s").WithLocation(7, 15),
+                // (8,9): warning CS8602: Possible dereference of a null reference.
+                //         s.ToString(); // warn 2
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "s").WithLocation(8, 9)
                 );
         }
 
         [Fact]
-        public void NotNullWhenFalse_MissingAttribute()
+        public void MethodWithoutParameter_WithNullableOut()
```

Comment:
Is this different from `MethodWithoutNullableParameter`?

Reply:
They are indeed duplicates. Thanks
Btw, I'm going to revert the `WithOut` => `Without` change, as `WithOut` is correct for those tests. It means "with out parameter..." not "without parameter...".

---
In reply to: [188459997](https://github.com/dotnet/roslyn/pull/26828#discussion_r188459997) [](ancestors = 188459997)